lib/super_digit.rb

```diff
@@ -1,15 +1,23 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n^2)
+# Space Complexity - O(n^2)
 def super_digit(n)
-   
+  return n.to_i if n.to_s.length == 1
+  digits = n.to_s.chars.map(&:to_i)
+  sum = 0
+  digits.each do |digit|
+    sum += digit
+  end
+  return sum if sum.to_s.length == 1
+  return super_digit(sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n^2)
+# Space Complexity - O(n^2)
 def refined_super_digit(n, k)
```

Comment:
This is a brute-force version of `refined_super_digit`, can you think of a dynamic programming approach.